src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -14,9 +15,18 @@ func (client *Client) initBalancer(cfg *ClientConfig) *balancer.Balancer {
 	dialers := make([]*balancer.Dialer, 0, len(cfg.FrontedServers)+len(cfg.ChainedServers))
 
 	log.Debugf("Adding %d domain fronted servers", len(cfg.FrontedServers))
+	var highestQOSFrontedDialer fronted.Dialer
+	highestQOS := -1
 	for _, s := range cfg.FrontedServers {
-		dialer := s.dialer(cfg.MasqueradeSets)
+		fd, dialer := s.dialer(cfg.MasqueradeSets)
 		dialers = append(dialers, dialer)
+		if dialer.QOS > highestQOS {
+			highestQOSFrontedDialer = fd
+		}
+	}
+
+	if highestQOSFrontedDialer != nil {
+		go determinePublicIp(highestQOSFrontedDialer)
 	}
```

Comment:
Done.  I made it debug since it doesn't actually prevent the functioning of the system, it just keeps us from getting a public ip.